AppCenter/AppCenter/Internals/Channel/MSChannelUnitDefault.m

```diff
@@ -371,30 +371,58 @@ - (void)flushQueue {
 
 - (void)checkPendingLogs {
 
-  // Skip checking pending logs if the channel is paused.
-  if (self.paused) {
-    return;
-  }
-
   // If the interval is default and we reached batchSizeLimit flush logs now.
-  if (self.configuration.flushInterval == kMSFlushIntervalDefault && self.itemsCount >= self.configuration.batchSizeLimit) {
+  if (!self.paused && self.configuration.flushInterval == kMSFlushIntervalDefault && self.itemsCount >= self.configuration.batchSizeLimit) {
     [self flushQueue];
   } else if (self.itemsCount > 0) {
     NSUInteger flushInterval = [self resolveFlushInterval];
-    if (flushInterval == 0) {
 
-      // If the interval is over, send all logs without any additional timers.
+    // Skip sending logs if the channel is paused.
+    if (self.paused) {
```

Comment:
Can be moved to 
```obj-c
else if (self.itemsCount >0 ) {
```
condition

Reply:
@Jamminroot No, it can't since `resolveFlushInterval` has to be executed in this case

Reply:
Yup, missed the fact that `resolveFlushInterval` modifies data.